### ISSUE TITLE
Any Plans to Support arm64 Images

### ISSUE THREAD

**Issue by timnolte** [WordPress/wpdev-docker-images#127]
I was starting to use the PHP image(https://hub.docker.com/r/wordpressdevelop/php) for plugin & theme development only to find that there are currently no arm64 versions of the images. Is this on anyone's radar? These images supply much more items that are required for development and CI/CD that the standard WordPress Docker images(https://hub.docker.com/_/wordpress) don't provider, like Xdebug.

**Comment by timnolte** [comment]
For the time being I've basically replicated the `php` image Dockerfile from here in my own `Dockerfile` for building on arm64. https://github.com/ndigitals/wp-dev-container/blob/feature/Initial-Setup/.devcontainer/Dockerfile

I'm working on creating Dev Container pre-built images that include a number of PHP & NodeJS versions for both amd64 & arm64 platforms. I'm able to create the arm64 images because I have native arm64 GitHub Action runners. If you need any assistance with anything needed to support arm64 image builds I'd be happy to provide any assistance.

**Comment by johnbillion** [comment]
See https://core.trac.wordpress.org/ticket/59930 for related discussion and a workaround

**Comment by timnolte** [comment]
Hmm, there is no workaround if I'm using these images which only provide amd64 images. Sure M# macs can use a level of emulation but the experience is very often slow. However, in my case I'm building arm64 Docker images on arm64/Linux, I can't use emulation to create final native arm64 images.

**Comment by ocean90** [comment]
The workflows should get updated to make use of the official Docker actions to [build multi-platform images](https://docs.docker.com/build/ci/github-actions/multi-platform/).

**Comment by desrosj** [comment]
The workaround in https://core.trac.wordpress.org/ticket/59930 works for the MySQL images in versions <= 5.7, but the PHP images we maintain here are only `amd64` and cause the following notice when starting the environment:

`! php The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`

I thought I had created an issue for switching over to the official Docker actions to build our images, but I think it was just a side discussion. I think that's the right way to go, and it opens the door to add support for more architectures as requested here.

It's on my list, though at a very low priority. If someone wants to create a PR switching the workflows over to use the recommended Docker actions, I'll happily review.

**Comment by timnolte** [comment]
Just a word of warning though that GitHub doesn't provide native arm64 runners. You have to use QEMU and those build take a huge amount of time. I ended up building my own image, actually a Dev Container prebuilt image, that is based on this image and is built on native arm64 GitHub Action Runners that I maintain. Just as a comparison it would take many hours to build this image with QEMU as compared to in the minutes with a native arm64 runner.

**Comment by desrosj** [comment]
> Just a word of warning though that GitHub doesn't provide native arm64 runners.

@timnolte Just to make sure I understand. Do you mean the GitHub Action runner images? It does look like there is a `macos-14` [image in beta that uses an M1 processor](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-private-repositories).

**Comment by timnolte** [comment]
@desrosj yes, I am talking about the GitHub Actions Runners. I know there has been something in the works at GitHub to make this available. I think the thing is that there is still a bit of complexity around generating the multiplatform images. I have some self-hosted arm64 runners that I use and it was a chore to get my multiplatform images created properly.

**Comment by GingerGeek** [comment]
Hi all, I have done a PR which builds arm64 images and publishes both amd64 and arm64 containers as part of a multi-platform image format

This is my first contribution to the WordPress ecosystem, I was wanting to contribute in core but was having frustrations with the devcontainer setup there which I wanted to use this image for. I had seen for both devcontainer and for this repo there was complaints around support mac/arm64 so I thought I'd give it a stab

**Comment by timnolte** [comment]
@GingerGeek so I had already done this and submitted a PR that uses a multi-platform dev container image that can be used for this. My PR was rejected for use of that image so I don't think you are going to have any traction doing this.

My image is here: https://github.com/ndigitals/wp-dev-container

**Comment by timnolte** [comment]
> @GingerGeek so I had already done this and submitted a PR that uses a multi-platform dev container image that can be used for this. My PR was rejected for use of that image so I don't think you are going to have any traction doing this.
> 
> My image is here: https://github.com/ndigitals/wp-dev-container

@GingerGeek disregard I was thinking this was a different repo. Your PR is to hopefully actually make official arm64 images available.

**Comment by GingerGeek** [comment]
Yes, my intention would be to suggest adopting these developer images for the devcontainer within the main Wordpress repo as it would provide better consistency between workflows. However there may be some additional plumbing required.

**Comment by ockham** [comment]
I have come across an issue that _might_ be due to the lack of an arm64 version of the `wordpressdevelop/php` Docker image when resetting my `wordpress-develop` environment the other day (including a `docker system prune --all --volumes`).

I’m using [Rancher Desktop](https://rancherdesktop.io/) (essentially a free Docker Desktop clone) with VZ as the virtualization layer (and virtiofs as mount type) on an Apple Silicone based machine. Crucially, I had to enable Rosetta in order to get the dev env server to work:

<img width="1244" alt="Image" src="https://github.com/user-attachments/assets/61d00cf9-6d54-4f3b-9537-c7c80416cfd1" />

_Without_ Rosetta enabled, the dev server wouldn’t start, even though it seemed like it did -- there wasn't any error message. However, there was simply nothing listening at `localhost:8889`. Running `npm run env:logs` then revealed an error with the php container:

```
php-1                | qemu-x86_64: QEMU internal SIGSEGV {code=MAPERR, addr=0xffff857c5384}
```

I first ran into this issue some 2-3 weeks ago. Since I got it to work by enabling Rosetta, I suspect that there might be some binary dependency compiled into the `wordpressdevelop/php` image that probably wasn't there before (?)

Anyway, just wanted to leave this here in case someone else runs into the same problem. As stated above, the short-term workaround is to enable Rosetta; in the long term, we should really provide arm64 versions of our Docker images.

**Comment by GingerGeek** [comment]
@ockham Yes that would be related to lack of arm64 images.

Could you try the images built here?

https://github.com/WordPress/wpdev-docker-images/pull/167

`gingergeek/wpdev-docker-images-php:latest` should be a drop in replacement for `wordpressdevelop/php` and support both arm64 and amd64

**Comment by ockham** [comment]
> Could you try the images built here?
> 
> [#167](https://github.com/WordPress/wpdev-docker-images/pull/167)

Done, see https://github.com/WordPress/wpdev-docker-images/pull/167#issuecomment-2979910041 🙂
